src/lib/pickles/step.mli

```diff
@@ -1,10 +1,13 @@
 open Pickles_types
 
 module Make
+    (Inductive_rule : Inductive_rule.Intf with type 'a proof = 'a Proof.t)
```

Comment:
It does not seem to be obvious right now as we keep Kimchi proofs explicitly in some parts of the code, but I am wondering if we can remove the type constraint `type 'a proof = 'a Proof.t` later.